data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/Main.java

```diff
@@ -108,26 +113,46 @@ public static void main(final String[] args) {
       )
     );
 
+    final var waitVerticle = new CountDownLatch(1);
     vertx.deployVerticle(verticle)
-      .onSuccess(v -> logger.info("receiver started"))
-      .onFailure(t -> logger.error("receiver not started", t));
+      .onSuccess(v -> {
+        logger.info("Receiver started");
+        waitVerticle.countDown();
+      })
+      .onFailure(t -> {
+        // This is a catastrophic failure, close the application
+        logger.error("Consumer deployer not started", t);
+        vertx.close(v -> System.exit(1));
+      });
+    waitVerticle.await(5, TimeUnit.SECONDS);
+
+    final var publisher = new ContractPublisher(vertx.eventBus(), ResourcesReconcilerMessageHandler.ADDRESS);
+    final var fs = FileSystems.getDefault().newWatchService();
+    var fw = new FileWatcher(fs, publisher, new File(env.getDataPlaneConfigFilePath()));
+
+    // Gracefully clean up resources.
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
Without makind complex abstractions, can we wrap part of this shutdown code somewhere (like a static method) and place it in core? I see some cloned code here